radix-transactions/src/model/ledger_transaction.rs

```diff
@@ -68,71 +107,167 @@ impl PreparedLedgerTransaction {
                         system_transaction_hash: t.system_transaction_hash(),
                     }
                 }
-                PreparedLedgerTransactionInner::UserV1(t) => TypedTransactionIdentifiers::User {
+                PreparedLedgerTransactionInner::User(t) => TypedTransactionIdentifiers::User {
                     intent_hash: t.transaction_intent_hash(),
                     signed_intent_hash: t.signed_transaction_intent_hash(),
                     notarized_transaction_hash: t.notarized_transaction_hash(),
                 },
-                PreparedLedgerTransactionInner::RoundUpdateV1(t) => {
+                PreparedLedgerTransactionInner::Validator(t) => {
                     TypedTransactionIdentifiers::RoundUpdateV1 {
                         round_update_hash: t.round_update_transaction_hash(),
                     }
                 }
-                PreparedLedgerTransactionInner::FlashV1(t) => {
+                PreparedLedgerTransactionInner::ProtocolUpdate(t) => {
                     TypedTransactionIdentifiers::FlashV1 {
                         flash_transaction_hash: t.flash_transaction_hash(),
                     }
                 }
-                PreparedLedgerTransactionInner::UserV2(t) => TypedTransactionIdentifiers::User {
-                    intent_hash: t.transaction_intent_hash(),
-                    signed_intent_hash: t.signed_transaction_intent_hash(),
-                    notarized_transaction_hash: t.notarized_transaction_hash(),
-                },
             },
         }
     }
+
+    fn validate(
+        self,
+        validator: &TransactionValidator,
+        accepted_kind: AcceptedLedgerTransactionKind,
+    ) -> Result<ValidatedLedgerTransaction, TransactionValidationError> {
+        let validated_inner = match self.inner {
+            PreparedLedgerTransactionInner::Genesis(t) => {
+                if !accepted_kind.permits_genesis() {
+                    return Err(TransactionValidationError::Other(
```

Comment:
looks like we could be more concrete than a catch-all "Other", e.g. `TransactionValidationError::DisallowedTransactionKind`, wdyt?